### PR TITLE
added xPath & methods for ChallengesPage & ChallengeComponent

### PR DESCRIPTION
**Pull request by janestadnytska**

### CODE REVIEW

**Review by Maksym637 — APPROVED** [review]

File: src/main/java/org/ssu/edu/teachua/ui/pages/challenges/ChallengesPage.java [hunk 1]
```diff
@@ -1,10 +1,114 @@
...
+    @FindBy(how = How.XPATH, using = ".//span[@class='ant-input-group-addon']")
+    private WebElement searchBtn;
+    @FindBy(how = How.XPATH, using = "(.//*[@class='ant-table-cell']//a)[1]")
+    private WebElement challengeId;
+    @FindBy(how = How.XPATH, using = "(.//*[@class='ant-table-cell']//a)[2]")
+    private WebElement challengeSortNumber;
+    @FindBy(how = How.XPATH, using = "(.//*[@class='ant-table-cell']//a)[3]")
+    private WebElement challengeName;
+    @FindBy(how = How.XPATH, using = "(.//*[@class='ant-table-cell']//a)[4]")
+    private WebElement challengeTitle;
```
Comment (lhalam) [inline src/main/java/org/ssu/edu/teachua/ui/pages/challenges/ChallengesPage.java:29]: delete this

File: src/main/java/org/ssu/edu/teachua/ui/pages/challenges/ChallengesPage.java [hunk 2]
```diff
@@ -1,10 +1,114 @@
...
+    @FindBy(how = How.XPATH, using = ".//*[@title='Previous Page']")
+    private WebElement previousPage;
+    @FindBy(how = How.XPATH, using = ".//*[@title='Next Page']")
+    private WebElement nextPage;
+    @FindBy(how = How.XPATH, using = ".//*[@title='Next 5 Pages']")
+    private WebElement stepOverNextFivePages;
+    @FindBy(how = How.XPATH, using = ".//*[contains(@class, 'level-0 editable-row')]")
+    private List<WebElement> challengeLocators;
```
Comment (lhalam) [inline src/main/java/org/ssu/edu/teachua/ui/pages/challenges/ChallengesPage.java:37]: move to get ChalengesTableItemComponent

File: src/main/java/org/ssu/edu/teachua/ui/components/challenge/ChallengeComponent.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.How;
+import org.ssu.edu.teachua.ui.base.BaseComponent;
+import org.ssu.edu.teachua.ui.pages.challenges.ChallengesPage;
+
+public class ChallengeComponent extends BaseComponent {
```
[the remark this answers is not in the document]
Comment (janestadnytska) [reply src/main/java/org/ssu/edu/teachua/ui/components/challenge/ChallengeComponent.java:11]: done

File: src/main/java/org/ssu/edu/teachua/ui/components/challenge/ChallengeComponent.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    public ChallengesPage clickChallengeTitle() {
+        challengeTitle.click();
+        return new ChallengesPage(driver);
+    }
+
+    public ChallengesPage clickEditButton() {
+        editBtn.click();
+        return new ChallengesPage(driver); //this
+    }
```
[the remark this answers is not in the document]
Comment (janestadnytska) [reply src/main/java/org/ssu/edu/teachua/ui/components/challenge/ChallengeComponent.java:66]: done

File: src/main/java/org/ssu/edu/teachua/ui/pages/challenges/ChallengesPage.java (hunk 1 above)
Comment (janestadnytska) [reply src/main/java/org/ssu/edu/teachua/ui/pages/challenges/ChallengesPage.java]: done

File: src/main/java/org/ssu/edu/teachua/ui/pages/challenges/ChallengesPage.java (hunk 2 above)
Comment (janestadnytska) [reply src/main/java/org/ssu/edu/teachua/ui/pages/challenges/ChallengesPage.java]: done

**Pull request merged**